packages/fuel-gauge/src/advanced-logging.test.ts

```diff
@@ -160,33 +148,135 @@ describe('Advanced Logging', () => {
+        ])
+        .call();
+
+      logs.forEach((log, i) => {
+        expect(JSON.stringify(log)).toBe(JSON.stringify(expectedLogs[i]));
+      });
+    });
+
+    it('when using ScriptTransactionRequest', async () => {
+      const callTest = await setupCallTest({ cache: false });
+      const configurable = await setupConfigurable({ cache: false });
+      const coverage = await setupCoverage({ cache: false });
+
+      const request = await callTest
+        .multiCall([
+          advancedLogContract.functions
+            .test_log_from_other_contract(10, otherLogId)
+            .addContracts([otherAdvancedLogContract]),
+          callTest.functions.boo(testStruct),
+          configurable.functions.echo_struct(),
+          coverage.functions.echo_str_8('fuelfuel'),
+        ])
+        .getTransactionRequest();
+
+      const { maxFee, gasUsed, requiredQuantities } = await provider.getTransactionCost(
+        request,
+        [],
+        {
+          resourcesOwner: wallet,
+        }
+      );
+
+      request.gasLimit = gasUsed;
+      request.gasPrice = minGasPrice;
+
+      await wallet.fund(request, requiredQuantities, maxFee);
+
+      const tx = await wallet.sendTransaction(request);
+
+      const { logs } = await tx.waitForResult();
+
+      expect(logs).toBeDefined();
+
+      logs?.forEach((log, i) => {
+        if (typeof log === 'object') {
+          expect(JSON.stringify(log)).toBe(JSON.stringify(expectedLogs[i]));
+        } else {
+          expect(log).toBe(expectedLogs[i]);
+        }
+      });
     });
   });
 
-  it('should decode logs from a script set to manually call other contracts', async () => {
+  describe('decode logs from a script set to manually call other contracts', () => {
     const { abiContents, binHexlified } = getFuelGaugeForcProject(
       FuelGaugeProjectsEnum.SCRIPT_CALL_CONTRACT
     );
 
-    const wallet = await generateTestWallet(provider, [[1_000]]);
-
-    const script = new Script(binHexlified, abiContents, wallet);
-
     const amount = Math.floor(Math.random() * 10) + 1;
 
-    const { logs } = await script.functions
-      .main(advancedLogId, otherLogId, amount)
-      .addContracts([advancedLogContract, otherAdvancedLogContract])
-      .call();
+    let wallet: WalletUnlocked;
 
-    expect(logs).toStrictEqual([
+    const expecedLogs = [
       'Hello from script',
       'Hello from main Contract',
       'Hello from other Contract',
       'Received value from main Contract:',
       amount,
-    ]);
+    ];
+
+    beforeAll(async () => {
+      wallet = await generateTestWallet(provider, [[1_000]]);
+    });
+
+    it('when using InvocationScope', async () => {
+      const script = new Script(binHexlified, abiContents, wallet);
+      const { logs } = await script.functions
+        .main(advancedLogId, otherLogId, amount)
+        .addContracts([advancedLogContract, otherAdvancedLogContract])
+        .call();
+
+      expect(expecedLogs).toStrictEqual(logs);
```

Comment:
```suggestion
      expect(logs).toStrictEqual(expecedLogs);
```

Reply:
There's a typo `expecedLogs` -> `expectedLogs`

Reply:
Save us from airdrop farmers and fix it :sweat_smile:

Reply:
@nedsalk you make a strong case 🤣